gtsam/hybrid/HybridFactor.h

```diff
@@ -41,6 +41,9 @@ KeyVector CollectKeys(const KeyVector &keys1, const KeyVector &keys2);
 DiscreteKeys CollectDiscreteKeys(const DiscreteKeys &key1,
                                  const DiscreteKeys &key2);
 
+/// Enum to help with categorizing hybrid factors.
+enum class HybridCategory { None, Discrete, Continuous, Hybrid };
```

Comment:
I like this a lot, but could we just make it an inner inside the HF class?

Reply:
Done. I renamed it to `Category` so it reads nicer as `HybridFactor::Category`.